### DISCUSSION TITLE
Experimental logs mode

### DISCUSSION THREAD

**Discussion by irevoire** [meilisearch/product#723] [Experimental features]
<table>
    <thead>
        <tr>
            <th>Feature name</th>
  			<th>Enable when launching</th>
  			<th>Description</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>Update the mode of the logs outputted in the console</td>
           <td>

```
./meilisearch --experimental-logs-mode human
```
  
</td>            
<td>Output logs as json</td>
        </tr>
    </tbody>
</table>

Meilisearch v1.7.0 (released on March 11, 2024) introduces a new cli flag to output logs as json.

## Experimental feature

Enabling the experimental feature will give you the possibility to output logs as json, which should help in case you're sending your logs to a log storage solution like quickwit, loki, elastic search, etc, ...

## Usage

Two modes are supported right now:
- `human`: Output logs in a human-readable format: `2024-02-13T17:28:11.427631Z  INFO actix_server::builder: starting 10 workers` for example
- `json`: Output logs in json: `{"timestamp":"2024-02-14T10:33:22.439880Z","level":"INFO","fields":{"message":"starting 10 workers"},"target":"actix_server::builder"}`

```bash
./meilisearch --experimental-logs-mode human
# OR
./meilisearch --experimental-logs-mode json
```

------

A few notes on the feature:
- When outputting json, all colors are removed, so if you were previously running a sanitizing pass to get rid of the shell formatting, you won't need it anymore
- There is currently no way to enable or disable this feature at runtime; you need to restart meilisearch to change its value

**Comment by macraig** [comment]
**Update:** [v1.7](https://github.com/meilisearch/meilisearch/releases/tag/v1.7.0) has been released 🐇
